File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
from math import sqrt
def solution(number, limit, power):
    answer = 1
    divisor = set()
    for i in range(2, number+1):
        tmp = 2
        for j in range(2, int(sqrt(i))+1):
            if (i == j):
                continue
            if i % j == 0:
                tmp += 2
        num = int(sqrt(i))
        if (num * num == i):
            tmp -= 1
        # print(tmp)
        if tmp > limit:
            answer += power
        else:
            answer += tmp
            
    
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 1;
        for (int i = 2; i <= number; i++) {
            int tmp = 2;
            int num = (int) Math.floor(Math.sqrt(i));
            for (int j = 2; j <= num; j++) {
                if (i % j == 0) {
                    tmp += 2;
                }
            }
            if (num * num == i) {
                tmp -= 1;
            }
            
            if (tmp > limit) {
                answer += power;
            } else {
                answer += tmp;
            }
        }
        return answer;
    }
}